control/tests/test_heal.py

```diff
@@ -1,36 +1,150 @@
+        },
+        {
+            'edx_id': '1',
+            'video_trans_status': 'Review Reject',
+            'video_trans_start': datetime.datetime.utcnow().replace(tzinfo=utc),
+            'video_active': True,
+            'inst_class': {}
+        },
+        {
+            'edx_id': '1',
+            'video_trans_status': 'Review Hold',
+            'video_trans_start': datetime.datetime.utcnow().replace(tzinfo=utc),
+            'video_active': True,
+            'inst_class': {}
+        },
+        {
+            'edx_id': '1',
+            'video_trans_status': 'Complete',
+            'video_trans_start': datetime.datetime.utcnow().replace(tzinfo=utc),
+            'video_active': False,
+            'inst_class': {}
+        },
+        {
+            'edx_id': '2',
+            'video_trans_status': 'Ingest',
+            'video_trans_start': datetime.datetime.utcnow().replace(tzinfo=utc),
+            'video_active': True,
+            'inst_class': {}
+        },
+        {
+            'edx_id': '1',
+            'video_trans_status': 'Corrupt File',
+            'video_trans_start': datetime.datetime.utcnow().replace(tzinfo=utc),
+            'video_active': True,
+            'inst_class': {}
+        }
+
+    )
+    @unpack
+    def test_determine_fault(self, edx_id, video_trans_status, video_active, inst_class, video_trans_start):
+        """
+        Tests that determine_fault works as expected.
+        """
+        video_instance = Video(
+            edx_id=edx_id,
+            video_trans_status=video_trans_status,
+            video_trans_start=video_trans_start,
+            video_active=video_active,
+            inst_class=Course()
+        )
+        encode_list = self.heal_instance.determine_fault(video_instance)
 
-    def test_encode_url(self):
-        H = VedaHeal()
-        H.discovery()
+        if video_instance.edx_id == '1':
+            self.assertEqual(encode_list, [])
+        elif video_instance.edx_id == '2':
+            for e in encode_list:
+                self.assertTrue(e in self.encode_list)
 
+    @data(
+        {
+            'uncompleted_encodes': [],
+            'expected_encodes': ['test_obj'],
+            'video_object': {
+                'edx_id': '1',
+                'video_trans_status': 'Complete',
+                'video_trans_start': datetime.datetime.utcnow().replace(tzinfo=utc),
+                'video_active': True,
+                'inst_class': {
+                    'review_proc': True
+                }
+            }
+        },
+        {
+            'uncompleted_encodes': ['test_obj'],
+            'expected_encodes': ['test_obj'],
+            'video_object': {
+                'edx_id': '2',
+                'video_trans_status': 'Ingest',
+                'video_trans_start': datetime.datetime.utcnow().replace(tzinfo=utc),
+                'video_active': True,
+                'inst_class': {
+                    'review_proc': True
```

Comment:
Can we have some data with review_proc False, and video_active False, The purpose is to complete the coverage by manipulating values of object.

Reply:
@muzaffaryousaf Yep - review_proc is beyond the scope of this function, but video_active is a good test.

Reply:
@muzaffaryousaf Actually, this just tests whether or not the list comparison is functioning as expected. The previous test tests for active/inactive video states. I'll update the docstring.